cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,282 @@
+            )
+
+    def get_consensus_jobs(self, task_id: int) -> None:
+        job_map = {}  # parent_job_id -> [(consensus_job_id, assignee)]
+        parent_jobs: dict[int, Job] = {}
+        for job in (
+            Job.objects.prefetch_related("segment", "parent_job", "assignee")
+            .filter(
+                segment__task_id=task_id,
+                type=JobType.CONSENSUS.value,
+                parent_job__stage=StageChoice.ANNOTATION.value,
+                parent_job__isnull=False,
+            )
+            .exclude(state=StateChoice.NEW.value)
+        ):
+            job_map.setdefault(job.parent_job_id, []).append((job.id, job.assignee))
+            parent_jobs.setdefault(job.parent_job_id, job.parent_job)
+
+        self.jobs = job_map
+        self.parent_jobs = list(parent_jobs.values())
+
+    def get_assignee_jobs_count(self) -> None:
+        assignee_jobs_count = {}
+        for assignees in self.jobs.values():
+            for _, assignee in assignees:
+                if assignee not in assignee_jobs_count:
+                    assignee_jobs_count[assignee] = 1
+                else:
+                    assignee_jobs_count[assignee] += 1
+        self.assignee_jobs_count = assignee_jobs_count
+
+    @staticmethod
+    def get_annotations(job_id: int) -> dm.Dataset:
+        return JobDataProvider(job_id).dm_dataset
+
+    def _merge_consensus_jobs(self, parent_job_id: int):
+        consensus_job_info = self.jobs.get(parent_job_id)
+        if not consensus_job_info:
+            raise ValidationError(f"No consensus jobs found for parent job {parent_job_id}")
+
+        consensus_job_ids = [consensus_job_id for consensus_job_id, _ in consensus_job_info]
+        assignees = [assignee for _, assignee in consensus_job_info]
+
+        consensus_job_data_providers = list(map(JobDataProvider, consensus_job_ids))
+        consensus_datasets = [
+            consensus_job_data_provider.dm_dataset
+            for consensus_job_data_provider in consensus_job_data_providers
+        ]
+
+        merged_dataset = self.merger(consensus_datasets)
+
+        assignee_report_data = generate_assignee_consensus_report(
+            consensus_job_ids,
+            assignees,
+            consensus_datasets,
+            self.merger.dataset_mean_consensus_score,
+        )
+
+        # delete the existing annotations in the job
+        patch_job_data(parent_job_id, None, PatchAction.DELETE)
+        # if we don't delete exising annotations, the imported annotations
+        # will be appended to the existing annotations, and thus updated annotation
+        # would have both existing + imported annotations, but we only want the
+        # imported annotations
+
+        parent_job_data_provider = JobDataProvider(parent_job_id)
+
+        # imports the annotations in the `parent_job.job_data` instance
+        import_dm_annotations(merged_dataset, parent_job_data_provider.job_data)
```

Comment:
After merging I get annotations with the `MANUAL` source. Could you try to update it to be `CONSENSUS` instead?

![image](https://github.com/user-attachments/assets/74a567d3-fe7b-49c0-8dae-670538520fa4)